### PR TITLE
Added support for java 11

### PR DESCRIPTION
**Pull request by lukidzi**
Build passed with openjdk version "11.0.3" locally

### CODE REVIEW

**Comment by druminski** [comment]
It would be nice to use [openjdk11 in travis](https://docs.travis-ci.com/user/languages/java/#using-java-10-and-later)

File: hermes-api/build.gradle [hunk 1]
```diff
@@ -9,6 +9,10 @@ dependencies {
     compile group: 'com.fasterxml.jackson.jaxrs', name: 'jackson-jaxrs-json-provider', version: versions.jackson
     compile group: 'com.google.guava', name: 'guava', version: versions.guava
     compile group: 'com.damnhandy', name: 'handy-uri-templates', version: '2.0.2'
+    compile group: 'javax.xml.bind', name: 'jaxb-api', version: '2.2.11'
```
Comment (druminski) [inline hermes-api/build.gradle]: Do we need all of these dependencies?

File: hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy [hunk 2]
```diff
@@ -129,7 +129,7 @@ class FilterChainSpec extends Specification {
         def order
 
         RecordingSubscriptionMessageFilterCompiler(String type, AtomicInteger counter) {
-            super(type, { m -> test(m) })
+            super(type, null)
```
Comment (druminski) [inline hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy]: Why did you change it to null?

Comment (lukidzi) [reply hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy]: With newer version of groovy-spock lib you cannot pass method of instance which hasn't been constructed yet.

File: hermes-api/build.gradle (hunk 1 above)
Comment (lukidzi) [reply hermes-api/build.gradle]: Those are java ee dependencies removed from jdk 11 and they are need in this package

File: hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy (hunk 2 above)
Comment (dankraw) [reply hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy]: so the 2nd argument wasn't used at all?

Comment (lukidzi) [reply hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/filtering/FilterChainSpec.groovy]: Yes it wasn't, this overrided method was called because  RecordingSubscriptionMessageFilterCompiler extends MessageFilter

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/ssl/SslContextFactory.java [hunk 3]
```diff
@@ -1,8 +1,11 @@
 package pl.allegro.tech.hermes.common.ssl;
 
 import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
 
 public interface SslContextFactory {
 
     SSLContext create();
+
+    TrustManager[] getTrustManagers();
```
Comment (dankraw) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/ssl/SslContextFactory.java]: As we talked already, we could change this interface, so it would have a single method:
`SSLContextHolder create()` or something similar... :)
and `SSLContextHolder` would hold both the `TrustManagers[]` and `SSLContext` instances. This way we explicitly say that these object always stick together and are created at the same time by the factory.

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/ssl/SSLContextHolder.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+public final class SSLContextHolder {
+
+    private final SSLContext sslContext;
+    private final TrustManager[] trustManagers;
+
+    public SSLContextHolder(SSLContext sslContext) {
```
Comment (dankraw) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/ssl/SSLContextHolder.java]: Maybe we should remove this constructor?

**Comment by dankraw** [comment]
@lukidzi 👏 👏
